app/src/interfaces/input-rich-text-html/useImage.ts

```diff
@@ -118,11 +116,7 @@ export default function useImage(
 	}
 
 	function onImageSelect(image: File) {
-		const fileExtension = image.type
-			? readableMimeType(image.type, true)
-			: readableMimeType(mime.getType(image.filename_download) as string, true);
-
-		const assetUrl = getPublicURL() + 'assets/' + image.id + '.' + fileExtension;
+		const assetUrl = getPublicURL() + 'assets/' + image.filename_disk;
```

Comment:
Even though we're not directly exposing this through the app/api, there's a world where folks have manually updated the filename_disk which means it no longer matches the `<id>.<ext>` format.. I'm wondering if we should play it safe here and make the url `'assets/' + image.id + <extension from filename_disk>`. In the assets endpoint, everything after the UUID is optional and ignored, so that should work for cases where filename_disk was manually overridden as well

Reply:
I made the change to only pull the extension from `filename_disk`. The current change makes the assumption that the `filename_disk` will always have a file extension. If this is a bad assumption, I can extract the old functionality out, and the new functionality into a method that checks if `filename_disk` is undefined, to use the old logic check. Otherwise I can check if the file `type` is `image/jpg` and compare it against `jpg/jpeg` and return the correct value.

Reply:
Hmm good point. The filename_disk is generated as UUID + ext by default, but can theoretically be overridden during upload. That in turn means that there's no guarantee that the extension actually exists.

Reply:
Would you like me to make an alteration to have a fallback to pull as it did previously with an additional check for `jpg/jpeg` values?

Reply:
I think we could do a check to see if there's a period in the filename_disk first `.includes('.')` and do the split+pop if true. Then fallback to the previous logic if not 🙂